week3/yourChoice/src/components/Content/Prefer.jsx

```diff
@@ -0,0 +1,222 @@
+import ContentWrapper from '../Layout/ContentWrapper';
+import ContentHeader from '../Layout/ContentHeader';
+import Result from '../Layout/Result';
+import H1 from '../UI/H1';
+
+const initialState = {
+  step: 0,
+  goNextBtn: false,
+  btnActive: '',
+  pickedArr: { 0: '', 1: '', 2: '' },
+};
+
+const reducerFn = (state, action) => {
+  switch (action.type) {
+    case 'TYPE_CLICKED':
+      return {
+        ...state,
+        goNextBtn: true,
+        btnActive: action.btnActive,
+        pickedArr: {...state.pickedArr, [state.step]: action.pickedArr},
+      };
+
+    case 'NEXT_BTN_CLICKED':
+      return {
+        ...state,
+        step: state.step + 1,
+        goNextBtn: action.goNextBtn,
+      };
+
+    case 'PREV_BTN_CLICKED':
+      if (state.step === 0) {
+        action.setGameHandler(false);
+        return { ...state };
+      } else {
+        return {
+          ...state,
+          btnActive: SONG_DATA[state.step - 1][state.pickedArr[state.step - 1]],
+          step: state.step - 1,
+          goNextBtn: true,
+        };
+      }
+
+    case 'RESULT_BTN_CLICKED':
+      return {
+        ...state,
+        step: state.step + 1,
+      };
+  }
+};
+
+// 취향대로 content
+const Prefer = (props) => {
+  const [gameState, dispatchGameState] = useReducer(reducerFn, initialState);
+
+  // 이미지 경로 매칭
+  const [imgSrc, setImgSrc] = useState('');
+
+  // 노래 제목 매칭
+  const [songTitle, setSongTitle] = useState('');
+
+  // 타입 선택 버튼
+  const onClickTypeBtn = (e) => {
+    dispatchGameState({
+      type: 'TYPE_CLICKED',
+      btnActive: e.target.value,
+      pickedArr: e.target.name
+    });
+  };
+
+  // 다음으로
+  const onClickNextBtn = () => {
+    dispatchGameState({ type: 'NEXT_BTN_CLICKED', goNextBtn: false });
+  };
+
+  // 이전으로
+  const onClickPrevBtn = () => {
+    dispatchGameState({
+      type: 'PREV_BTN_CLICKED',
+      setGameHandler: props.setGameHandler
+    });
+  };
+
+  // 결과보기 버튼
+  const onClickResultBtn = () => {
+    let result = [];
+
+    for (const [key, value] of Object.entries(gameState.pickedArr)) {
+      result.push(value);
+    }
+
+    let resultImgSrc = `src/assets/img/p${result[0]}_${result[1]}_${result[2]}.jpeg`;
+    dispatchGameState({ type: 'RESULT_BTN_CLICKED' });
+
+    setSongTitle(SONG_TITLE[result[0]][result[1]][result[2]]);
+    setImgSrc(resultImgSrc);
+  };
+
+  return (
+    <>
+      <ContentWrapper>
+        <ContentHeader>
+          <H1>{CONTENT_HEADER[gameState.step]}</H1>
+        </ContentHeader>
+
+        {gameState.step !== 3 ? <StepNum>{gameState.step + 1}/3</StepNum> : ''}
+
+        <ChooseContainer>
+          {gameState.step !== 3 ? (
+            SONG_DATA[gameState.step].map((item, idx) => {
+              return (
+                <SelectBtn
+                  key={idx}
+                  name={idx}
+                  value={item}
+                  className={item == gameState.btnActive ? 'active' : ''}
+                  onClick={onClickTypeBtn}
+                >
```

Comment:
아래에 `StepGoNextButton`에서는 `goNextBtn` 값을 styledComponent에게 prop으로 내려주었는데, 여기서는 `btnActive` 여부를 별도의 prop이 아닌 className으로 지정해서 스타일링해준 이유가 혹시 있을까요!? 

따로 className을 지정해주지 않아도 아래처럼 **boolean 값을 prop에 담아** styled component로 넘겨주고, 해당 값에 따라 삼항연산자로 background-color를 지정해주면 `&.action`을 별도로 스타일링하는 코드를 싹 지워줘도 된답니다ㅎㅎ 참고하세요!!
 ```suggestion
                <SelectBtn
                  key={idx}
                  name={idx}
                  value={item}
                  isBtnActive={item == gameState.btnActive}
                  onClick={onClickTypeBtn}
                >
```